test/uk/gov/hmrc/nrs/retrieval/connectors/NonrepRetrievalConnectorSpec.scala

```diff
@@ -16,55 +16,120 @@
 
 package uk.gov.hmrc.nrs.retrieval.connectors
 
-import org.mockito.ArgumentMatchers
+import com.google.inject.{AbstractModule, Guice, Injector}
+import org.apache.pekko.util.ByteString
 import org.mockito.ArgumentMatchers.*
 import org.mockito.Mockito.when
-import uk.gov.hmrc.http.client.{HttpClientV2, RequestBuilder}
-import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import play.api.Environment
+import play.api.libs.ws.WSClient
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.nrs.retrieval.UnitSpec
+import uk.gov.hmrc.nrs.retrieval.config.AppConfig
 
-import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.client.RequestBuilder
+import uk.gov.hmrc.http.HttpResponse
+
+import scala.concurrent.{ExecutionContext, Future}
+import play.api.libs.ws.DefaultBodyWritables.writeableOf_String
+import play.api.libs.ws.DefaultBodyWritables.writeableOf_Bytes
 
 class NonrepRetrievalConnectorSpec extends UnitSpec:
-  private val mockHttpClientV2           = mock[HttpClientV2]
-  private val mockHttpResponse           = mock[HttpResponse]
-  private val mockRequestBuilder         = mock[RequestBuilder]
-  given mockHeaderCarrier: HeaderCarrier = mock[HeaderCarrier]
+  private val mockWsHttp                     = mock[HttpClientV2]
+  private val mockEnvironment                = mock[Environment]
+  private val mockAppConfig                  = mock[AppConfig]
+  private val mockWSClient                   = mock[WSClient]
+  private val mockHttpResponse               = mock[HttpResponse]
+  private val requestBuilder: RequestBuilder = mock[RequestBuilder]
+
+  implicit val mockHeaderCarrier: HeaderCarrier = mock[HeaderCarrier]
+
+  private val testModule = new AbstractModule:
```

Comment:
is this the platform pattern for testing connectors we discussed?

Reply:
The two line:
```
bind(classOf[HttpClientV2]).toInstance(mockWsHttp)
private val connector: NonrepRetrievalConnector = injector.getInstance(classOf[NonrepRetrievalConnector])
```
are doing the same thing

Unfortunately  the first mock is for httpClientV2, when calling search() then it will call httpClientV2.get(...)  (or or really mock.get(...))   and so that will need to be mocked with another mock that can handle the next line ( mock.transform(...) ) etc.
I don't think HttpClientV2 was designed for mocking